File: packages/shared/src/authzen.ts
// OpenID AuthZEN 1.0 Access Evaluation 요청/응답(lazyfga-9).

export interface EvaluationRequest {
  subject: { type: string; id: string };
  action: { name: string };
  resource: { type: string; id: string };
  context?: Record<string, unknown>;
  /** reason=true면 응답 context.reason에 설명 부착(lazyfga-11). */
  options?: { reason?: boolean };
}

export interface EvaluationResponse {
  decision: boolean;
  context?: Record<string, unknown>;
}

File: apps/api-go/internal/contract/authzen.go
package contract

// OpenID AuthZEN 1.0 Access Evaluation 요청/응답(LFGA-9).

type EvalSubject struct {
	Type string `json:"type"`
	ID   string `json:"id"`
}

type EvalAction struct {
	Name string `json:"name"`
}

type EvalResource struct {
	Type string `json:"type"`
	ID   string `json:"id"`
}

// EvalOptions: reason=true면 응답 context.reason에 설명을 부착(LFGA-11).
type EvalOptions struct {
	Reason *bool `json:"reason,omitempty"`
}

type EvaluationRequest struct {
	Subject  EvalSubject    `json:"subject"`
	Action   EvalAction     `json:"action"`
	Resource EvalResource   `json:"resource"`
	Context  map[string]any `json:"context,omitempty"`
	Options  *EvalOptions   `json:"options,omitempty"`
}

type EvaluationResponse struct {
	Decision bool           `json:"decision"`
	Context  map[string]any `json:"context,omitempty"`
}
